### PR TITLE
Handle usedGas

### PR DESCRIPTION
**Pull request by Sajma27**
# Summary
A description of the changes proposed in the pull request.

Fixes #number

## Checklist
- [ ] Change is small and easy to review (please split big changes into multiple PRs)
- [ ] Change is consistent with architecture
- [ ] Change is consistent with test architecture
- [ ] Change is consistent with naming conventions
- [ ] New code is covered with tests
- [ ] Tests related to old code are updated
- [ ] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-relayer/src/integration/ethereum/DeploymentExecutor.ts [hunk 1]
```diff
@@ -24,8 +24,12 @@ export class DeploymentExecutor implements IExecutor<Deployment> {
       const {hash, wait, gasPrice} = transactionResponse;
       ensureNotFalsy(hash, TransactionHashNotFound);
       await this.deploymentRepository.markAsPending(deploymentHash, hash!, gasPrice.toString());
-      await wait();
-      await this.deploymentRepository.setState(deploymentHash, 'Success');
+      const {gasUsed} = await wait();
+      if (gasUsed) {
+        await this.deploymentRepository.markAsSuccess(deploymentHash, gasUsed.toString());
+      } else {
+        throw new Error('No gas used');
```
Comment (xitronix) [inline universal-login-relayer/src/integration/ethereum/DeploymentExecutor.ts:31]: Gas used not found in transaction receipt

Comment (xitronix) [reply universal-login-relayer/src/integration/ethereum/DeploymentExecutor.ts:31]: We should introduce new error or just handle it in else like this.deploymentRepository.markAsError(deploymentHash, 'Gas used not found in transaction receipt)

File: universal-login-relayer/test/core/services/execution/QueueService.test.ts [hunk 2]
```diff
@@ -69,7 +69,7 @@ describe('UNIT: Queue Service', async () => {
     await queueMemoryStore.addMessage(messageHash);
     await waitExpect(() => expect(executeSpy).to.be.calledOnce);
     await waitExpect(() => expect(wait).to.be.calledOnce);
-    expect(onTransactionMined).to.be.calledImmediatelyAfter(wait);
+    expect(onTransactionMined).to.be.calledAfter(wait);
```
Comment (xitronix) [inline universal-login-relayer/test/core/services/execution/QueueService.test.ts:72]: Why did it stop work?

File: universal-login-relayer/src/core/utils/errors.ts [hunk 3]
```diff
@@ -23,7 +24,7 @@ type ErrorType =
   'InvalidApiKey' |
   'EnsNameTaken' |
   'UnauthorisedAddress' |
-  'InvalidRefundReceiver';
+  'InvalidRefundReceiver' ;
```
Comment (JustynaBroniszewska) [inline universal-login-relayer/src/core/utils/errors.ts:27]:
```suggestion
  'InvalidRefundReceiver';
```

**Pull request merged**